lib/services/api/api_service.ts

```diff
@@ -234,4 +256,56 @@ export class SNApiService extends PureService {
 
     return response;
   }
+
+  private async checkForExpiredAccessToken(errorResponse: HttpResponse) {
```

Comment:
`checkForExpiredAccessToken` doesn't imply there are side-effects (refresh session). The two actions taken here should be separated.